strapi/strapi
# Fix sanity check for @casl/ability upgrade #12232

### What does it do?
- Works on solving issue #12232. 
- Normalizes the codebase for the BREAKING CHANGES introduced in v5.1.0 of @casl/ability 

### Why is it needed?
- npm v8 installations are not working because of @casl/ability v4 that does not support it and makes strapi fail on the installation.

Describe the issue you are solving.

### How to test it?

- Unit and e2e tests were all passed to check for the changes made in this PR. 

To assert for a green light on an npm v8 installation (#12232) I used "verdaccio" to mimic a public npm registry locally so I can have a registry to publish forth and pull back. After that I went with the standard worflow on how to create a strapi app but with npm (--use-npm flag). 

The steps I used:
- `npm i -g verdaccio` 
- `npm i -g npm` - install npm@latest
- `verdaccio` - opens the registry and starts listening at http://localhost:4873
- `npm set registry http://localhost:4873`  ​
- `cd path/to/root/of/strapi/monorepo/`
- `lerna publish --force-publish --no-git-tag-version --no-push` - publishes strapi to verdaccio
- `cd ..` 
- `npx create-strapi-app my-strapi-app --quickstart --use-npm` - install strapi using npm pulling from verdaccio 
 `App is instaling ...`
 
Remember to put your npm registry back to npm when finished
 
### Related issue(s)/PR(s)

#12232

### Aditional Info

There are three commits that merge into one: 
- "HEAD~2" has all the code changes for the source files.
- "HEAD^" has comments related to the modifications in "HEAD~2".
- "HEAD" wipes out all the comments introduced in "HEAD^" and leaves an exact copy of "HEAD~2". 

### RFC
A ci test could be added to run on a relaxed schedule to test against package managers as well. Currently, tests are running against node versions and databases but package managers are a source of problems as well.
